modules/vmss/main.tf

```diff
@@ -1,259 +1,370 @@
+            domain_name_label = nic.value.pip_domain_name_label
           }
         }
       }
     }
   }
 
-
-  boot_diagnostics {
-    storage_account_uri = var.diagnostics_storage_uri
-  }
+  boot_diagnostics { storage_account_uri = var.virtual_machine_scale_set.diagnostics_storage_uri }
 
   identity {
-    type = "SystemAssigned" # (Required) The type of Managed Identity which should be assigned to the Linux Virtual Machine Scale Set. Possible values are SystemAssigned, UserAssigned and SystemAssigned, UserAssigned.
-  }
-
-  source_image_id = var.custom_image_id
-
-  source_image_reference {
-    publisher = var.use_custom_image ? null : var.img_publisher
-    offer     = var.use_custom_image ? null : var.img_offer
-    sku       = var.use_custom_image ? null : var.img_sku
-    version   = var.use_custom_image ? null : var.img_version
+    type         = var.virtual_machine_scale_set.identity_type
+    identity_ids = var.virtual_machine_scale_set.identity_ids
   }
 
-  os_disk {
-    caching                = "ReadWrite"
-    disk_encryption_set_id = var.disk_encryption_set_id #  The Disk Encryption Set must have the Reader Role Assignment scoped on the Key Vault - in addition to an Access Policy to the Key Vault.
-    storage_account_type   = var.storage_account_type
-  }
+  tags = var.tags
+}
 
-  dynamic "plan" {
-    for_each = var.enable_plan ? ["one"] : []
+locals {
+  # this loop will pull out all `window_minutes`-like properties from the scaling rules
+  # into one map that can be fed into the `pdt_time` module
+  profile_time_windows_flat = flatten([
+    for profile in var.autoscaling_profiles : [
+      for rule in profile.scale_rules : [
+        [
+          for k, v in rule.scale_out_config :
+          {
+            name  = "${profile.name}-${replace(lower(rule.name), " ", "_")}-scale_out-${k}"
+            value = v
+          }
+          if strcontains(k, "window_minutes")
+        ],
+        [
+          for k, v in rule.scale_in_config :
+          {
+            name  = "${profile.name}-${replace(lower(rule.name), " ", "_")}-scale_in-${k}"
+            value = v
+          }
+          if strcontains(k, "window_minutes") && v != null
+        ]
+      ]
+    ]
+  ])
+  profile_time_windows = { for v in local.profile_time_windows_flat : v.name => v.value }
+}
 
-    content {
-      name      = var.img_sku
-      publisher = var.img_publisher
-      product   = var.img_offer
-    }
-  }
+module "ptd_time" {
+  source   = "./dt_string_converter"
+  for_each = local.profile_time_windows
+  time     = each.value
 }
 
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_autoscale_setting
 resource "azurerm_monitor_autoscale_setting" "this" {
-  count = length(var.autoscale_metrics) > 0 ? 1 : 0
+  count = length(var.autoscaling_profiles) > 0 ? 1 : 0
 
-  name                = "${var.name}-autoscale"
+  name                = "${var.name}-autoscale-settings"
```

Comment:
Can we avoid hardcoding fragments of names for PIP (`"${var.name}-autoscale-settings"`)?